functions/logWorkAPI.js

```diff
@@ -31,74 +32,88 @@ exports.logWork = functions.https.onRequest((request, response) => {
+                const updatePromise = db.collection("Organizations").doc(documentID).collection("Tasks").doc(taskId).update({
+                    LogWorkTotalTime: logWorkTotalTime,
+                    WorkDone: workDone,
+                    Status: status,
+                    CompletionDate: completiondate,
+                });
+                return Promise.resolve(updatePromise);
             });
-            return Promise.resolve(updatePromise);
-        });
-        const promise2 = db.collection("RawData").doc("AppDetails").get().then((doc) => {
-            totalCompletedTask = parseInt(doc.data().TotalCompletedTask);
-            totalUnCompletedTask = parseInt(doc.data().TotalUnCompletedTask);
-            if (status === "Completed") {
-                totalCompletedTask = totalCompletedTask + 1;
-                totalUnCompletedTask = totalUnCompletedTask - 1;
-            }
-            const updateStatus = db.collection("RawData").doc("AppDetails").update({
-                TotalCompletedTask: totalCompletedTask,
-                TotalUnCompletedTask: totalUnCompletedTask,
+
+
+            const promise2 = db.collection("Organizations").doc(documentID).collection("RawData").doc("AppDetails").get().then((doc) => {
+                totalCompletedTask = parseInt(doc.data().TotalCompletedTask);
+                totalUnCompletedTask = parseInt(doc.data().TotalUnCompletedTask);
+                if (status === "Completed") {
+                    totalCompletedTask = totalCompletedTask + 1;
+                    totalUnCompletedTask = totalUnCompletedTask - 1;
+                }
+                const updateStatus = db.collection("Organizations").doc(documentID).collection("RawData").doc("AppDetails").update({
+                    TotalCompletedTask: totalCompletedTask,
+                    TotalUnCompletedTask: totalUnCompletedTask,
+                });
+                return Promise.resolve(updateStatus);
             });
-            return Promise.resolve(updateStatus);
-        });
-        const promise3 = db.collection("Main").doc(fullSprintId).get().then((doc) => {
-            totalCompletedTask = parseInt(doc.data().TotalCompletedTask);
-            totalUnCompletedTask = parseInt(doc.data().TotalUnCompletedTask);
-
-            if (status === "Completed") {
-                totalCompletedTask = totalCompletedTask + 1;
-                totalUnCompletedTask = totalUnCompletedTask - 1;
-            }
-            const updateSprintstatus = db.collection("Main").doc(fullSprintId).update({
-                TotalCompletedTask: totalCompletedTask,
-                TotalUnCompletedTask: totalUnCompletedTask,
+
+            const promise3 = db.collection("Organizations").doc(documentID).collection("Sprints").doc(fullSprintId).get().then((doc) => {
+                totalCompletedTask = parseInt(doc.data().TotalCompletedTask);
+                totalUnCompletedTask = parseInt(doc.data().TotalUnCompletedTask);
+
```

Comment:
I think these two promises should happen only when status is changed to completed

Reply:
Yes but then this is the method that was previously used in the older api